src/MSBuild/LiveLogger/LiveLogger.cs

```diff
@@ -134,16 +136,19 @@ private void eventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
             // Get project id
             int id = e.BuildEventContext!.ProjectInstanceId;
 
-            // If id does not exist...
-            if (!projects.ContainsKey(id))
+            // Add or update project
+            ProjectNode node = new(e)
             {
-                // Add project
-                ProjectNode node = new(e)
-                {
-                    ShouldRerender = true,
-                };
-                projects[id] = node;
-                UpdateFooter();
+                ShouldRerender = true,
+            };
+            projects[id] = node;
+            UpdateFooter();
+
+            if (e.TargetNames?.Contains("Restore") == true)
```

Comment:
I think it's worth it to explicitly specify `OrdinalIgnoreCase` here even if netfx makes us do it the ugly way

```suggestion
            if (e.TargetNames?.IndexOf("Restore", StringComparison.OrdinalIgnoreCase) > 0)
```